gradio/events.py

```diff
@@ -264,3 +265,34 @@ def __init__(self):
         """
         This event is triggered when the user releases the mouse on this component (e.g. when the user releases the slider). This method can be used when this component is in a Gradio Blocks.
         """
+
+
+@document("*select", inherit=True)
+class Selectable(EventListener):
+    def __init__(self):
+        self.selectable: bool = False
+        self.select = EventListenerMethod(
+            self, "select", callback=lambda: setattr(self, "selectable", True)
+        )
+        """
+        This event is triggered when the user selects from within the Component.
+        This event has EventData of type gradio.SelectEvent that carries information, accessible through SelectEvent.index and SelectEvent.value.
+        See EventData documentation on how to use this event data.
+        """
+
+
+class SelectEvent(EventData):
```

Comment:
I would find it more intuitive to call this class `SelectData` instead of `SelectEvent`, as it just holds the data related to the selection

Reply:
agreed.